broker/src/test/java/io/camunda/zeebe/broker/clustering/topology/ClusterTopologyTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.clustering.topology;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.broker.clustering.topology.MemberState.State;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ClusterTopologyTest {
+  @Test
+  void canInitializeClusterWithPreExistingMembers() {
+    // given
+    final var expectedTopology =
+        new ClusterTopology(
+            0,
+            Map.of(
+                member(1),
+                withActivePartition(0, 1),
+                member(2),
+                withActivePartition(0, 2),
+                member(3),
+                withActivePartition(0, 3)),
+            new ClusterChangePlan());
+    // when
+    final var topology =
+        ClusterTopology.init()
+            .addMember(
+                member(1), MemberState.initializeAsActive(Map.of(1, PartitionState.active(1))))
+            .addMember(
+                member(2), MemberState.initializeAsActive(Map.of(2, PartitionState.active(1))))
+            .addMember(
+                member(3), MemberState.initializeAsActive(Map.of(3, PartitionState.active(1))));
+
+    // then
+    assertThat(topology).isEqualTo(expectedTopology);
+  }
+
+  @Test
+  void shouldMergeConcurrentUpdatesToMembers() {
```

Comment:
:thought_balloon: Some sort of property based testing would be cool here...

Reply:
:smile: I thought about, but figured out might be too much at this stage. But definitely this is suitable for property tests. Can do it as a follow up.

Reply:
Yeah I thought the same, more of a note for later